File: app/src/main/java/com/dolphkon/wanandroidclient/ui/LoginContract.java
package com.dolphkon.wanandroidclient.ui;

import com.dolphkon.httplib.base.BaseView;
import com.dolphkon.httplib.base.IDisposable;
import com.dolphkon.wanandroidclient.bean.RegisterResp;

/**
 * ****************************************************
 * Project: WanandroidClient
 * PackageName: com.dolphkon.wanandroidclient.ui
 * ClassName：
 * Author: kongdexi
 * Date: 2020/10/29 11:52
 * Description:TODO
 * *****************************************************
 */
public interface LoginContract {
   interface View extends BaseView{
    void  updateRegisterView(RegisterResp registerResp);
    void  login(RegisterResp registerResp);
   }

    interface Presenter extends IDisposable{
        void  register(String account,String password,String repassword);
        void  login(String account,String password);
    }

}

File: app/src/main/java/com/dolphkon/wanandroidclient/ui/LoginContract.kt
package com.dolphkon.wanandroidclient.ui

import com.dolphkon.httplib.base.BaseView
import com.dolphkon.httplib.base.IPresenter

/**
 * ****************************************************
 * Project: WanandroidClient
 * PackageName: com.dolphkon.wanandroidclient.ui
 * ClassName：
 * Author: kongdexi
 * Date: 2020/10/29 11:52
 * Description:TODO
 * *****************************************************
 */
interface LoginContract {
    interface View : BaseView<Any?> {
        fun updateRegisterView()
    }

    interface Presenter : IPresenter<View?> {
        fun register(account: String?, password: String?, repassword: String?)
    }
}